---
marp: true
size: 4:3
theme: default
---

## Strand Orientation Bias Detector (SOBDetector) to remove artifacts from sequencing data of formalin fixed samples. 

---

by analyzing the orientation of paired end sequences that are spanning the variantsâ€™ locations.

The
most prominent of those is the deamination of cytosines, which introduces
CFT/GFA mutations (Oh et al., 2015). 


As a measure of the strand orientation, we defined the strand orientation bias
(SOB) score, as the following:
SOB = (#F1R2alt - #F2R1alt)/(#F1R2alt + #F1R2alt')
where #ofF1R2alt is the number of alternate allele supporting reads that are in the
forward 1st, reverse 2nd orientation.
SOB score lies in the [-1,1] interval, and the closer it is to the two extrema, the
higher the probability, that the corresponding variant is a sequencing artifact.

https://github.com/mikdio/SOBDetector

## Benchmark of tools for CNV detection from NGS panel data in a genetic diagnostics context
Five tools (DECoN, CoNVaDING, panelcn.MOPS, ExomeDepth and CODEX2) were tested against four genetic diagnostics datasets (495 samples, 231 CNVs)

(ICR96, panelcnDataset, In-house Miseq and In-HouseHiSeq)

+ The dataset ICR96 exon CNV validation series (Mahamdallie et al., 2017) was downloaded from the
European Genome-phenome Archive (EGA) (EGAD00001003335) and contained 96 samples captured
with TruSight Cancer Panel v2 (100 genes) and sequenced in a single HiSeq lane with 101-base pairedend reads. 
+  panelcnDataset (Povysil et al., 2017) was also downloaded from EGA (EGAS00001002481)
and had 170 samples captured using TruSight Cancer Panel (94 genes, Illumina) and sequenced on a
MiSeq instrument with 151-base paired-end reads. s. Only 161 out of the 170 samples were used in this
work and 9 were removed because they presented alterations out of the scope of this benchmark: 5
presented CNVs smaller than an exon (IBK9, IBK23, IBK67, IBK153, IBK166) and 4 contained ALUs
insertions instead of CNVs (IBK141, IBK142, IBK143, IBK151). IBK141 ALU insertion was not
reported in the original publication but identified in this work (Supplementary File 1). 