+ Copy-Number Variants Detection by Low-Pass Whole-Genome Sequencing
  + DOI: 10.1002/cphg.43
+ A Robust Approach for Blind Detection of Balanced Chromosomal Rearrangements with Whole-Genome Low-Coverage Sequencing
  + DOI: 10.1002/cphg.51
+ WisecondorX: improved copy number detection for routine shallow whole-genome sequencing
  + 旧版本：https://github.com/VUmcCGP/wisecondor
+ AMYCNE: Confident copy number assessment using whole genome sequencing data
+ MetaCNV - a consensus approach to infer accurate copy numbers from low coverage data
  +  ReadDepth v0.9.8, which applies a read depth approach, predicts mid-sized (104–106 bp) and larger deleted segments (> 106 bp) more accurately than other callers, whereas amplifications were more accurately predicted by SVDetect v1.3, a representative of the hybrid approach.
  + ReadDepth SVDetect CNVnator
  + https://bitbucket.org/sonnhammergroup/metacnv/src/master/
+ CNAtra: An analytical and visualization tool for hierarchical CNA discovery of large-scale and focal copy number alterations in low-coverage cancer genomes (Khalil et. al., BMC Bioinformatics, 2020)
  + https://github.com/AISKhalil/CNAtra
  + https://pubmed.ncbi.nlm.nih.gov/32299346/
  + MATLAB based
+ Performance of copy number variants detection based on whole-genome sequencing by DNBSEQ platforms
  + 数据集
  + https://link.springer.com/article/10.1186/s12859-020-03859-x
+ An open resource for accurately benchmarking small variant and reference calls
  + x
+ Copy number variant detection with low-coverage wholegenome sequencing is a viable alternative to the traditional array-CGH
  + PMID: 30061371 DOI: 10.1136/jmedgenet-2018-105272
+ Copy number variant detection with low-coverage whole-genome sequencing is a viable replacement for the traditional array-CGH
  + Marcel Kucharík 1,2,*, Jaroslav Budiš 1,2,3, Michaela Hýblová 4, Gabriel Minárik 4, and Tomáš Szemes 1,2,5 
+ Characterizing sensitivity and coverage of clinical WGS as a diagnostic test for genetic disorders
+ https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-021-00841-x
+ Comprehensively benchmarking applications for detecting copy number variation
  + Le Zhang  ,Wanyu Bai ,Na Yuan ,Zhenglin Du  
  + https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007069
  + 通过灵敏度，特异性和计算要求对应用程序进行了基准测试。以DGV黄金标准变体作为标准数据集，我们用10X到50X的测序深度评估了具有真实测序数据的十个应用程序。在基准的十种方法中，LUMPY在每个测序深度的高灵敏度和特异性方面均表现最佳。出于高度特异性的目的，Canvas也是一个不错的选择。如果首选高灵敏度，则CNVnator和RDXplorer是更好的选择。此外，CNVnator和GROM-RD在处理低深度测序数据方面也表现出色。
  + overlap of more than 50% 
  
----
+ Whole-genome sequencing analysis of copy number variation (CNV) using low-coverage and paired-end strategies is efficient and outperforms array-based CNV analysis
  + Evaluation of copy number variation detection between high-resolution array CGH and low-coverage short-insert and mate-pair whole-genome sequencing
  +  Bo Zhou, doi: https://doi.org/10.1101/192310
  + https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP118709
  + https://www.ncbi.nlm.nih.gov//sra/?term=SRP118709
  +  Furthermore, we show that WGS (at 1x coverage) is able to detect
17 all seven GS deletion-CNVs >100 kb in NA12878 whereas only one is detected by most
18 arrays. 
  + Lastly, we show that the much larger 15 Mbp Cri-du-chat deletion can be readily
19 detected with short-insert paired-end WGS at even just 1x coverage.
  + LiangD,PengY,LvW,DengL,ZhangY,LiH,YangP,ZhangJ,SongZ,XuG, Cram DS, Wu L. Copy number variation sequencing for comprehensive diagnosis of chromosome disease syndromes. J Mol Diagn 2014;16:519–26.
Liu S, Song L, Cram DS, Xiong L, Wang K, Wu R, Liu J, Deng K, Jia B, Zhong M, Yang F. Traditional karyotyping vs copy number variation sequencing for
detection of chromosomal abnormalities associated with spontaneous miscarriage. Ultrasound Obstet Gynecol 2015;46:472–7. DongZ,ZhangJ,HuP,ChenH,XuJ,TianQ,MengL,YeY,WangJ,ZhangM, LiY,WangH,YuS,ChenF,XieJ,JiangH,WangW,WaiChoyK,XuZ.Low- pass whole-genome sequencing in clinical cytogenetics: a validated approach. Genet Med 2016;18:940–8.

----
Haraksingh RR, Abyzov A, Urban AE. Comprehensive performance comparison
of high-resolution array platforms for genome-wide Copy Number Variation (CNV) analysis in humans. BMC Genomics 2017;18:321.

----
+ ichorNCA
