By default, FusionCatcher its the Bowtie aligner for finding candidate fusion genes. 

--skip-blat','--skip-star', or '--skip-bowtie2'

 '--aligners=blat,star,bowtie2'

https://github.com/ndaniel/fusioncatcher/issues/84#issuecomment-487368403


| **Fusion\_description** | **Description** |
|:------------------------|:----------------|
| **1000genomes**             | fusion gene has been seen in a healthy sample. It has been found in [RNA-seq data from some samples from 1000 genomes project](http://dx.doi.org/10.1371/journal.pone.0104567). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **18cancers**              | fusion gene found in a RNA-seq dataset of 18 types of cancers from 600 tumor samples published [here](http://dx.doi.org/10.1073/pnas.1606220113). |
| **adjacent**           | both genes forming the fusion are adjacent on the genome (i.e. same strand and there is no other genes situated between them on the same strand)|
| **antisense**           | one or both genes is a gene coding for [antisense RNA](http://en.wikipedia.org/wiki/Antisense_RNA)|
| **banned**              | fusion gene is on a list of known false positive fusion genes. These were found with very strong supporting data in healthy samples (i.e. it showed up in file final-list\_candidate\_fusion\_genes.txt). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **bodymap2**            | fusion gene is on a list of known false positive fusion genes. It has been found in healthy human samples collected from 16 organs from  [Illumina BodyMap2 RNA-seq database](http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-513/). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **cacg**                | known conjoined genes (that is fusion genes found in samples from healthy patients) from the [CACG](http://cgc.kribb.re.kr/map/) database (please see CACG database for more information). *A candidate fusion gene having this label has a very high probability of being a false positive in case that one looks for fusion genes specific to a disease.*|
| **cell\_lines**         | known fusion gene from paper: C. Klijn et al., A comprehensive transcriptional portrait of human cancer cell lines, Nature Biotechnology, Dec. 2014, [DOI:10.1038/nbt.3080](http://dx.doi.org/10.1038/nbt.3080) |
| **ccle**         | known fusion gene found in human cancer cell lines that are in CCLE (Cancer Cell Line Encyclopedia); known fusions from [DepMap portal](https://depmap.org/portal/download/) |
| **ccle3**         | known fusion gene found in human cancer cell lines that are in CCLE (Cancer Cell Line Encyclopedia); known fusions from from paper: Vellichirammal et al., Pan-Cancer Analysis Reveals the Diverse Landscape of Novel Sense and Antisense Fusion Transcripts,  Mol. Ther. Nucleic Acids, 2020, [DOI:10.1016/j.omtn.2020.01.023](https://doi.org/10.1016/j.omtn.2020.01.023) |
| **cgp**                 | known fusion gene from the [CGP](http://www.sanger.ac.uk/genetics/CGP/Census/) database |
| **chimerdb2**           | known fusion gene from the [ChimerDB 2](http://ercsb.ewha.ac.kr/FusionGene/) database|
| **chimerdb3kb**           | known fusion gene from the [ChimerDB 3 KB (literature curration)](http://ercsb.ewha.ac.kr/FusionGene/) database |
| **chimerdb3pub**           | known fusion gene from the [ChimerDB 3 PUB (PubMed articles)](http://ercsb.ewha.ac.kr/FusionGene/) database |
| **chimerdb3seq**           | known fusion gene from the [ChimerDB 3 SEQ (TCGA)](http://ercsb.ewha.ac.kr/FusionGene/) database |
| **conjoing**            | known conjoined genes (that is fusion genes found in samples from healthy patients) from the [ConjoinG](http://metasystems.riken.jp/conjoing/) database (please use ConjoinG database for more information regarding the fusion gene). *A candidate fusion gene having this label has a very high probability of being a false positive in case that one looks for fusion genes specific to a disease.* |
| **cortex**            | fusion gene is on a list of known false positive fusion genes. It has been found in healthy human brains (BA9 prefrontal cortex) [here](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1679648). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **cosmic**              | known fusion gene from the [COSMIC](http://cancer.sanger.ac.uk/cancergenome/projects/cosmic/) database (please use COSMIC database for more information regarding the fusion gene) |
| **distance1000bp**      | both genes are on the same strand and they are less than 1 000 bp apart. *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **distance100kbp**      | both genes are on the same strand and they are less than 100 000 bp apart. *A candidate fusion gene having this label has a higher probability than expected of being a false positive.* |
| **distance10kbp**       | both genes are on the same strand and they are less than 10 000 bp apart. *A candidate fusion gene having this label has a higher probability than expected of being a false positive.* |
| **duplicates**          | both genes involved in the fusion gene are [paralog](http://en.wikipedia.org/wiki/Paralog#Paralogy) for each other. For more see [Duplicated Genes Database (DGD) database](http://dgd.genouest.org/) . *A candidate fusion gene having this label has a higher probability than expected of being a false positive.* |
| **exon-exon**          | the fusion junction point is exactly at the known exon's borders of both genes forming the candidate fusion |
| **ensembl\_fully\_overlapping** | the genes forming the fusion gene are fully overlapping according to Ensembl database. *A candidate fusion gene having this label has a very high probability of being a false positive.*|
| **ensembl\_partially\_overlapping** | the genes forming the fusion gene are partially overlapping (on same strand or on different strands) according the Ensembl database. *A candidate fusion gene having this label has a good probability of being a false positive.</i> </font> |
| **ensembl\_same\_strand\_overlapping** | the genes forming the fusion gene are fully/partially overlapping and are both on the same strand according to Ensembl database. *A candidate fusion gene having this label has a very high probability of being a false positive (this is most likely and alternative splicing event).</i> </font> |
| **fragments** | the genes forming the fusion are supported by only and only one fragment of RNA. *A candidate fusion gene having this label has a medium probability of being a false positive.*|
| **gliomas**              | fusion gene found in a RNA-seq dataset of 272 glioblastoms published [here](http://dx.doi.org/10.1101/gr.165126.113). |
| **gtex**             | fusion gene has been seen in a healthy sample. It has been found in [GTEx database](http://www.gtexportal.org/home/) of healthy tissues (thru [FusionAnnotator](https://github.com/FusionAnnotator/FusionAnnotator)). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **healthy**             | fusion gene has been seen in a healthy sample. These have been found in healthy samples but the support for them is less strong (i.e. paired reads were found to map on both genes but no fusion junction was found) than in the case of **banned** label (i.e. it showed up in file preliminary list of candidate fusion genes). Also genes which have some degree of sequence similarity may show up marked like this.*A candidate fusion gene having this label has a small probability of being a false positive in case that one looks for fusion genes specific to a disease.* |
| **hpa**             | fusion gene has been seen in a healthy sample. It has been found in [RNA-seq database of 27 healthy tissues](http://dx.doi.org/10.1074/mcp.M113.035600). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **known**       | fusion gene which has been previously reported or published in scientific articles/reports/books/abstracts/databases indexed by [Google](http://www.google.com/), [Google Scholar](http://scholar.google.com/), [PubMed](http://www.ncbi.nlm.nih.gov/pubmed), etc. This label has only the role to answer with YES or NO the question "has ever before a given (candidate) fusion gene been published or reported?". This label does not have in anyway the role to provide the original references to the original scientific articles/reports/books/abstracts/databases for a given fusion gene. |
| **lincrna**             | one or both genes is a [lincRNA](http://en.wikipedia.org/wiki/Long_non-coding_RNA) |
| **matched-normal**      | candidate fusion gene (which is supported by paired reads mapping on both genes and also by reads mapping on the junction point) was found also in the matched normal sample given as input to the command line option '--normal' |
| **metazoa**             | one or both genes is a metazoa\_srp gene [Metazia\_srp](http://www.genecards.org/index.php?path=/Search/keyword/metazoa_srp) |
| **mirna**               | one or both genes is a [miRNA](http://en.wikipedia.org/wiki/MicroRNA) |
| **mt**                  | one or both genes are situated on [mitochondrion](http://en.wikipedia.org/wiki/Mitochondrion). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **mX** (where X is a number) | count of pairs of reads supporting the fusion (**excluding** the mutimapping reads). |
| **m0** | There are no pairs of non-multi-mapping reads supporting the fusion. Basically, there are supporting pairs of reads but all of them map also in some other places on genome (that is their mappings on genome are not unique). |
| **non\_cancer\_tissues**   | fusion gene which has been previously reported/found in non-cancer tissues and cell lines in [Babiceanu et al, Recurrent chimeric fusion RNAs in non-cancer tissues and cells, Nucl. Acids Res. 2016](http://nar.oxfordjournals.org/content/early/2016/02/01/nar.gkw032.abstract). These are considered as non-somatic mutation and therefore they may be skipped and not reported. |
| **non\_tumor\_cells**   | fusion gene which has been previously reported/found in non-tumor cell lines, like for example HEK293. These are considered as non-somatic mutation and therefore may be skipped and not reported. |
| **no\_protein** | one or both genes have no known protein product |
| **oesophagus**              | fusion gene found in a oesophageal tumors from TCGA samples, which are published [here](http://dx.doi.org/10.1038/nature20805). |
| **oncogene**            | one gene or both genes are a known [oncogene](http://en.wikipedia.org/wiki/Oncogene) according to [ONGENE database](https://doi.org/10.1016/j.jgg.2016.12.004) |
| **cancer**            | one gene or both genes are cancer associated according to [Cancer Gene database](http://www.bushmanlab.org/links/genelists) |
| **tumor**            | one gene or both genes are proto-oncogene or tumor suppresor gene according to [UniProt database](http://www.uniprot.org) |
| **short\_repeats** | the sequence of the fusion junction contains a highly repetitive region containing repeating short sequences or polyA/C/G/T (detected using kmer = 2) . *A candidate fusion gene having this label has a good probability of being a false positive.</i> </font> |
| **long\_repeats** | the sequence of the fusion junction contains a highly repetitive region containing repeating long sequences or polyA/C/G/T (detected using kmer = 9) . *A candidate fusion gene having this label has a good probability of being a false positive.</i> </font> |
| **pair\_pseudo\_genes** | one gene is the other's [pseudogene](http://en.wikipedia.org/wiki/Pseudogene). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **pancreases**           | known fusion gene found in pancreatic tumors from article: P. Bailey et al., Genomic analyses identify molecular subtypes of pancreatic cancer, Nature, Feb. 2016, http://dx.doi.org/110.1038/nature16965 |
| **paralogs**            | both genes involved in the fusion gene are  [paralog](http://en.wikipedia.org/wiki/Paralog#Paralogy) for each other (most likely this is a false positive fusion gene). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **multi**            | one of the genes of both have multi-mapping reads mapping (which map simultaneously also on other gene/genes |
| **partial-matched-normal** | candidate fusion gene (which is supported by paired reads mapping on both genes **but** _no_ reads were found which map on the junction point) was found also in the matched normal sample given as input to the command line option '--normal'. This is much weaker than **matched-normal**. |
| **prostates**           | known fusion gene found in 150 prostate tumors RNAs from paper: D. Robison et al, Integrative Clinical Genomics of Advanced Prostate Cancer, Cell, Vol. 161, May 2015, http://dx.doi.org/10.1016/j.cell.2015.05.001 |
| **pseudogene**          | one or both of the genes is a [pseudogene](http://en.wikipedia.org/wiki/Pseudogene) |
| **readthrough**         | the fusion gene is a readthrough event (that is both genes forming the fusion are on the same strand and there is no known gene situated in between); Please notice, that many of readthrough fusion genes might be false positive fusion genes due to errors in Ensembl database annotation (for example, one gene is annotated in Ensembl database as two separate genes). *A candidate fusion gene having this label has a high probability of being a false positive.* |
| **refseq\_fully\_overlapping** | the genes forming the fusion gene are fully overlapping according to RefSeq NCBI database. *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **refseq\_partially\_overlapping** | the genes forming the fusion gene are partially overlapping (on same strand or on different strands) according the RefSeq NCBI. *A candidate fusion gene having this label has a good probability of being a false positive.</i> </font> |
| **refseq\_same\_strand\_overlapping** | the genes forming the fusion gene are fully/partially overlapping and are both on the same strand according to RefSeq NCBI database. *A candidate fusion gene having this label has a very high probability of being a false positive (this is most likely and alternative splicing event).</i> </font> |
| **ribosomal**  | one or both gene is a gene encoding for [ribosomal protein](http://en.wikipedia.org/wiki/Ribosomal_protein) |
| **rrna**                | one or both genes is a [rRNA](http://en.wikipedia.org/wiki/Ribosomal_RNA).  *A candidate fusion gene having this label has a very high probability of being a false positive.*|
| **short\_distance**     | both genes are on the same strand and they are less than X bp apart, where X is set using the option '--dist-fusion' and by default it is 200 000 bp. *A candidate fusion gene having this label has a higher probability than expected of being a false positive.* |
| **similar\_reads**      | both genes have the same reads which map simultaneously on both of them (this is an indicator of how similar are the sequences of both genes; ideally this should be zero or as close to zero as possible for a real fusion). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **similar\_symbols**    | both genes have the same or very similar gene names (for example: RP11ADF.1 and RP11ADF.2). *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **snorna**              | one or both genes is a [snoRNA](http://en.wikipedia.org/wiki/Small_nucleolar_RNA) |
| **snrna**               | one or both genes is a [snRNA](http://en.wikipedia.org/wiki/Small_nuclear_RNA) |
| **tcga**                | known fusion gene from the [TCGA](https://tcga-data.nci.nih.gov/tcga/) database (please use Google for more information regarding the fusion gene) |
| **tcga-normal**                | known fusion gene from healthy samples from [TCGA](https://tcga-data.nci.nih.gov/tcga/) database (please use Google for more information regarding the fusion gene) |
| **tcga-cancer**                | known fusion gene from tumor samples from [TCGA](https://tcga-data.nci.nih.gov/tcga/) database (please use Google for more information regarding the fusion gene) |
| **tcga2**                | known fusion gene from the [TCGA](https://tcga-data.nci.nih.gov/tcga/) database (please use Google for more information regarding the fusion gene) |
| **tcga3**                | known fusion gene from the [TCGA](https://tcga-data.nci.nih.gov/tcga/) database (please use Google for more information regarding the fusion gene) |
| **ticdb**               | known fusion gene from the [TICdb](http://www.unav.es/genetica/TICdb/) database (please use TICdb database for more information regarding the fusion gene) |
| **trna**                | one or both genes is a [tRNA](http://en.wikipedia.org/wiki/Transfer_RNA) |
| **ucsc\_fully\_overlapping** | the genes forming the fusion gene are fully overlapping according to UCSC database. *A candidate fusion gene having this label has a very high probability of being a false positive.* |
| **ucsc\_partially\_overlapping** | the genes forming the fusion gene are partially overlapping (on same strand or on different strands) according the UCSC database.  *A candidate fusion gene having this label has a good probability of being a false positive.</i> </font> |
| **ucsc\_same\_strand\_overlapping** | the genes forming the fusion gene are fully/partially overlapping and are both on the same strand according to UCSC database. *A candidate fusion gene having this label has a very high probability of being a false positive (this is most likely and alternative splicing event).</i> </font> |
| **yrna**                | one or both genes is a [Y RNA](http://en.wikipedia.org/wiki/Y_RNA) |
