https://biopython.org/wiki/GFF_Parsing

## 获取intron区域
https://www.biostars.org/p/112251/#314840

```r
library(GenomicFeatures)

## make TxDb from GTF file 
txdb <- makeTxDbFromGFF('Homo_sapiens.GRCh38.93.gtf')

## get intron information
all.introns <- intronicParts(txdb)

## subset for introns greater than 5 kb (side-note: there are no introns longer than 5000 kbp in the human hg38 genome)
subset.introns <- all.introns[width(all.introns) > 5000]

## look at the number of unique gene names present
##(what i do is unlist some metadata... convert it to strings and then count the unique gene_name occurrences) I agree its a bit hard to read...

length(unique(as.character(unlist(mcols(subset.introns)$gene_id))))
19469

> write.table(all.introns, "all.introns.txt", sep = "\t", quote = F, row.names=F)
```
https://www.biostars.org/p/334830/#334833