002_CoverageCalculatorPy https://github.com/AWGL/CoverageCalculatorPy


## 对于cnv分析的几点思考：
#### 鉴定断裂点
+ CNVkit uses off-target reads between genes to detect CNV breakpoints, if your targeted sequencing protocol uses hybrid capture. In most of your cases this should help place the breakpoint correctly in the intergenic region instead of at the edge of the next targeted exon. Otherwise the CNVkit pipeline is conceptually similar to the workflow you're probably using.
+ If your protocol is targeted amplicon sequencing instead, and there are no off-target sequencing reads to improve your estimates, then try OncoCNV. After running CBS via DNAcopy, it uses another statistical test at the gene level to remove spurious breakpoints. I think this might trim the CNV to the gene of interest in your problematic cases.
+ The R package PSCBS, which wraps DNAcopy, has a procedure for estimating confidence intervals around breakpoints, which would help you detect breakpoints with questionable positioning near gene boundaries (if you don't mind some additional programming).

## 肿瘤拷贝数数据中的异方差校正
理想情况下，可以从整个基因组中均匀地采样DNA片段，但是Illumina测序仪倾向于具有非常高或很低GC含量的偏差。 GC含量是测序读段中G和C碱基对的数量。采样基因组中的其他偏差也可能发生。但是，通常通过对来自同一患者的配对正常组织样品进行测序来解决。然后，正常对照样本具有相同的偏倚，因此，当您计算肿瘤相对于正常对照的对数比时，偏倚就会抵消。但是，我之所以写这篇文章，是因为我有不幸的经验，无法消除那些偏见[2]。图2显示了引起的问题。它根据GC含量绘制log2（肿瘤/对照）。通常，这看起来很平坦，但是就GC含量而言，明显存在异方差，这是不希望的。为了更清楚地显示异方差，图3显示了0.005 GC含量的窗口中的方差估计。

DNA fragments would ideally be uniformly sampled from throughout the genome, but Illumina sequencing machines (for example) tend to have a bias with very high or low GC content. GC content is the number G and C basepairs in the sequenced read. Other biases in sampling the genome may also occur. However, this is usually dealt with by sequencing a paired normal tissue sample from the same patient. The normal control sample then has those same biases, so when you compute a log ratio of the tumor over normal control, the biases cancel out. However, the reason why I am writing this post is I had the unfortunate experience with having data where those biases did not cancel out [2]. Figure 2 shows the issue that caused. It plots the log2(tumor/control) against the GC content. Normally this would appear flat, but there is clear heteroskedasticity with respect to GC content, which was undesirable. To more clearly show the heteroskedasticity, Figure 3 shows the variance estimate in windows of 0.005 GC content.

https://murphycj.github.io/heteroskedasticity-correction/
