[TOC]

#
https://anjingwd.github.io/AnJingwd.github.io/2017/08/12/参考基因组数据下载/
# 2. Common
## 2.1. Ucsc(参考序列文件推荐下载这个)
进入UCSC后选择Downloads > Genome Data > Human > hg19 Full Dataset，
下载完成后需要解压文件并将所有染色体序列文件拼接整合成一个完整的hg38.fa文件
```
http://genome.ucsc.edu/cgi-bin/hgTables

http://genome.ucsc.edu/cgi-bin/hgTables?hgsid=682987339_Q4QiJkTbj7alIOKSnfjFgXGdrGKS&clade=mammal&org=&db=hg19&hgta_group=genes&hgta_track=knownGene&hgta_table=knownGene&hgta_regionType=genome&position=&hgta_outputType=primaryTable&hgta_outFileName=
```

```
wegt http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz ##hg19
wget http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/chromFa.tar.gz ##hg38

tar -zxvf chromFa.tar.gz
cd chroms
cat *.fa >hg19.fa
```

## 2.2. Gencode（基因注释文件下载）
(http://www.gencodegenes.org/)下载需要的注释文件，进入网站后选择Data > Human > GRCh37-mapped Releases
```
ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/

axel ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_28/GRCh37_mapping/gencode.v28lift37.annotation.gtf.gz
axel ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_28/GRCh37_mapping/gencode.v28lift37.annotation.gff3.gz
```

## 2.3. Ensembl
```
http://asia.ensembl.org/info/data/ftp/index.html

ftp://ftp.ensembl.org/pub/

ftp://ftp.ensembl.org/pub/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh38.87.chr.gtf.gz 

ftp://ftp.ensembl.org/pub/current_gtf/homo_sapiens/Homo_sapiens.GRCh38.90.gtf.gz## hg38

```
Gencode vs. Ensembl:

1. X/Y染色体上PAR区域的注释：该区域的注释在Gencode中X染色体和Y染色体各注释一次，而在Ensembl文件中，只在X染色体进行了注释

> PAR region(Pseudoautosomal region)：该区域是X和Y染色体的同源序列，因为这上面的任何基因可以和常染色体基因一样正常遗传而得名

2. Gencode的第九列，也就是attribute那一列，有一些其他额外的tags，这些tags是Ensebl所没有的

## 2.4. Refseq
```
ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/latest_assembly_versions/

wget ftp://ftp.ncbi.nlm.nih.gov/genomes/Homo_sapiens/GFF/ref_GRCh38.p7_top_level.gff3.gz          ## hg38
wget ftp://ftp.ncbi.nlm.nih.gov/genomes/Homo_sapiens/ARCHIVE/BUILD.37.3/GFF/ref_GRCh37.p5_top_level.gff3.gz    ## hg19
ftp://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh37_latest/refseq_identifiers/ ##hg19
```


```
##hg19：

for i in $(seq 1 22) X Y M;
do wget ftp://ftp.ncbi.nlm.nih.gov/genomes/Homo_sapiens/ARCHIVE/BUILD.37.3/Assembled_chromosomes/seq/hs_ref_GRCh37.p5_chr${i}.fa.gz; 
gzip -d hs_ref_GRCh37.p5_chr${i}.fa.gz;
cat hs_ref_GRCh37.p5_chr${i}.fa.gz >>hg19.fa;
sleep 10s;
done;

##hg38：
同理，只要找到下载地址，改动一下即可。
### ftp://ftp.ncbi.nlm.nih.gov/genomes/Homo_sapiens/Assembled_chromosomes/seq/hs_ref_GRCh38.p7_chr${i}.fa.gz
```
1000g:
```
http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/
```


#### Ref_Info
https://blog.csdn.net/tanzuozhev/article/details/78861704




### Reference genome patches
https://janbio.home.blog/2018/11/27/reference-genome-patches/

# 3. OMIM
OMIM，“0nline MendelianInheritance in Man”

https://www.omim.org

https://omim.org/static/omim/data/mim2gene.txt



## genemap2.txt说明
```
'Charcot-Marie-Tooth disease, axonal, type 2A2A, 609260 (3), Autosomal dominant; Charcot-Marie-Tooth disease, axonal, type 2A2B, 617087 (3), Autosomal recessive; Hereditary motor and sensory neuropathy VIA, 601152 (3), Autosomal dominant'

# Each Phenotype is followed by its MIM number, if different from
# that of the locus, preceded by a comma
# Phenotype mapping key in parentheses follows the phenotype MIM
# number (explanation below).
# Allelic disorders are separated by a semi-colon following the
# phenotype mapping key.
# Inheritance for the phenotype follows the phenotype mapping key
# preceded by a ,<space>
```
```
Allelic disorders, phenotype_MIMnumber(Phenotype mapping key), Inheritance
```
##


> http://vlambda.com/wz_x5GNF7ezSN.html

# 4. CADD

# 5. ExAC
+ Exome Aggregation Consortium (ExAC)
+ Consists of 60,706 unrelated individuals that were exome sequenced
+ Various disease-specific and population genetic studies
+ Search for any human gene or variant or region
+ 

https://raw.githubusercontent.com/zhezhangsh/rchive/master/R/ParseOmim.r

# 6. Cosmic
https://cancer.sanger.ac.uk/census

## 6. 
https://zhuanlan.zhihu.com/p/

# 7. dbSNP
NCBI Assay ID(ss)

Reference SNP ID(rs)
对于每一个提交到dbSNP数据库的SNP位点, 首先会赋予一个唯一的ss ID。 由于不同研究结构提交的SNP会存在冗余，提取SNP位点上下游区域的序列，比对参考基因组，如果多个ss ID 比对上相同的位置，说明这几个SNP位点是冗余的，会赋予一个新的reference SNP ID, 以rs开头。

ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606/VCF/

#### Ref_Info

## dbNSFP(非同义变异功能预测数据库）
所有非同义SNP位点的功能预测、保守性以及人口频率信息（如果可用）
![image](https://github.com/xiucz/pics/blob/master/dbNSFP.png?raw=true)

https://github.com/jmzeng1314/bioinformatics123/blob/master/appendix/database.md

https://anjingwd.github.io/AnJingwd.github.io/2017/08/12/参考基因组数据下载/