<!-- TOC -->

- [LOH](#loh)
    - [Install](#install)
- [](#)
- [reporter](#reporter)
- [pyloa](#pyloa)
- [-HD](#-hd)
    - [Install](#install-1)
    - [Running](#running)
- [](#-1)
- [snp2](#snp2)
- [scan](#scan)
- [x](#x)
- [attes](#attes)
- [Athlon](#athlon)
- [-HD](#-hd-1)
- [Optitype](#optitype)
- [miner](#miner)
- [POLYSOLVER](#polysolver)
- [seq2](#seq2)
- [PT](#pt)
- [-VBSeq](#-vbseq)
- [https://github.com/uci-cbcl/-bind](#httpsgithubcomuci-cbcl-bind)

<!-- /TOC -->

每个个体表达6个主要组织相容性复合体(MHC) I类等位基因，由位于6号染色体两个同源拷贝上的3个基因(HLA-A、HLA-B和HLA-C)编码。



基因，位于6号染色体上短臂上，长约4000Kb。是目前所知人体最复杂的遗传多态性系统，有几十个基因座位，每个基因座位又有几十个等位基因，且呈共显性表达。由于MHC基因位于同一条染色体上，其多基因座位上的基因型组合相对稳定，很少发生同源染色体间交换，这就构成了以单元型（HAPLOTYPE，即在同一条染色体上紧密连锁的一系列等位基因的特殊组合）为特征的遗传。按中国人常见的A座位基因有13个，B座位基因有30个计算，可组成的单元型约有13×30＝390种之多。

https://www.nature.com/articles/nbt.3344


使用WES数据对HLA分型的难点有两个：

1. 人类参考基因组的每一个HLA基因都有一个单一的序列，很可能会歪曲个体的真实等位基因，从而导致次优比对。
2. HLA基因富含GC碱基，因此由于捕获和扩增效率较低，测序覆盖率较低，测序错误增加，从而进一步降低了比对率。


## LOH
https://mp.weixin.qq.com/s/IgdAyckTTUeVMCUNUxrbww
### Install
```
conda install -c bioconda jellyfish
conda install -c conda-forge jellyfish
```
###

```
/lustre/Work/zhangbo/software/Anaconda3/envs/loh/bin/Rscript /lustre/Work/zhangbo/tmp/exon_more//loh/loh/LOHscript.R \
--patientId example \
--outputDir output \
--Path /lustre/Work/zhangbo/tmp/exon_more//loh/loh/example-file/s \
--fastaLoc /lustre/Work/zhangbo/tmp/exon_more//loh/loh/data/example.patient.Fasta.fa \
--exonLoc /lustre/Work/zhangbo/tmp/exon_more//loh/loh/data/.dat \
--CopyNumLoc /lustre/Work/zhangbo/tmp/exon_more//loh/loh/example-file/solutions.txt
```

## reporter

> Beside -A,B,C, DRB1, DQB1 etc I would like to analyse DRA and KIR(KIR3DL1) genes.

## pyloa
https://github.com/felixfan/Py

## -HD
+ version: v1.2.0.1
+ depth: 133x
官网地址：https://www.genome.med.kyoto-u.ac.jp/-HD/


### Install
```
# bowtie2
axel  https://github.com/BenLangmead/bowtie2/releases/download/v2.3.4.2/bowtie2-2.3.4.2-linux-x86_64.zip

# 
https://www.genome.med.kyoto-u.ac.jp/-HD/

#
update.dictionary.sh
```
###　Running
```
/lustre/Work/zhangbo/software/hd.1.2.0.1/bin/hd.sh \
-t 6 \
-m 100 \
-c 0.95 \
-f /lustre/Work/zhangbo/software/hd.1.2.0.1/freq_data/ \
$fastq_1 $fastq_2 \
/lustre/Work/zhangbo/software/hd.1.2.0.1/_gene.split.txt \
/lustre/Work/zhangbo/software/hd.1.2.0.1/dictionary/ \
$sampleid \
output_dir

```
###

## snp2
https://jiawen.zd200572.com/431.html

https://jiawen.zd200572.com/123.html

http://www.bioconductor.org/packages/devel/bioc/html/HIBAG.html

## scan
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1671-3

+ scan v2 uses Suffix Array (SA) method instead of BWA-mem for alignment with IMGT/ Database.
+ 

## x
+ https://github.com/humanlongevity/

对A、B、C、DPB1、DRB1进行分型。


## attes
https://github.com/cliu32/attes

## Athlon

## -HD
可以对classI、ClassII、ClassIII全部进行分型。
https://www.genome.med.kyoto-u.ac.jp/-HD/

## Optitype

## miner
http://www.bcgsc.ca/platform/bioinfo/software/miner


## seq2
+ RNA-Seq
+ https://bitbucket.org/sebastian_boegel/seq2
+ https://genomemedicine.biomedcentral.com/articles/10.1186/gm403
```
wget https://bitbucket.org/sebastian_boegel/seq2/get/seq2_v2.2.tar.gz
tar -zxvf seq2_v2.2.tar.gz
```

## PT
+ https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-15-325
+ 

## -VBSeq
http://nagasakilab.csml.org//

 
Allelelist.txt: ftp://ftp.ebi.ac.uk/pub/databases/ipd/imgt//Allelelist.txt

_all.fasta: ftp://ftp.ebi.ac.uk/pub/databases/ipd/imgt//fasta/*gen.fasta


https://www.zd200572.com/2017/09/10/x/

## https://github.com/uci-cbcl/-bind



https://github.com/mskcc/loh

https://github.com/humanlongevity/

http://fred-2.github.io/

https://github.com/FRED-2/OptiType

https://github.com/NCBI-Hackathons/HLAClustRView