<!-- TOC -->

- [VCF](#vcf)
- [插入片段](#插入片段)
- [过滤](#过滤)
- [benchmark data](#benchmark-data)

<!-- /TOC -->

Most 1000 Genomes SVs are deletions because these are easier to detect in low coverage sequencing data.[https://www.nature.com/articles/nature15394]

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5382922/

## VCF
5.4 Specifying complex rearrangements with breakends

An arbitrary rearrangement event can be summarized as a set of novel adjacencies. Each adjacency ties together 2 breakends. The two breakends at either end of a novel adjacency are called mates.
There is one line of VCF (i.e. one record) for each of the two breakends in a novel adjacency. A breakend record is identified with the tag “SYTYPE=BND” in the INFO field. The **REF field** of a breakend record indicates a base or sequence <font color="red">s</font> of bases beginning at position POS, as in all VCF records. The ALT field of a breakend record indicates a replacement for s. This “breakend replacement” has three parts:

1. The string t that replaces places s. The string t may be an extended version of s if some novel bases are inserted
during the formation of the novel adjacency.
2. The position p of the mate breakend, indicated by a string of the form “chr:pos”. This is the location of the
first mapped base in the piece being joined at this novel adjacency.
3. The direction that the joined sequence continues in, starting from p. This is indicated by the orientation of
square brackets surrounding p.
These 3 elements are combined in 4 possible ways to create the ALT. In each of the 4 cases, the assertion is that **s** is replaced with t, and then some piece starting at position p is joined to t. The cases are:

```
REF ALT Meaning
s t[p[ piece extending to the right of p is joined after t
s t]p] reverse comp piece extending left of p is joined after t
s ]p]t piece extending to the left of p is joined before t
s [p[t reverse comp piece extending right of p is joined before t
```
```
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO
2       321681  bnd     W       G       G]17:198982]    6       PASS    SVTYPE=BND
2       321682  bnd     V       T       ]13:123456]T    6       PASS    SVTYPE=BND
13      123456  bnd     U       C       C[2:321682[     6       PASS    SVTYPE=BND
13      123457  bnd     X       A       [17:198983[A    6       PASS    SVTYPE=BND
17      198982  bnd     Y       A       A]2:321681]     6       PASS    SVTYPE=BND
17      198983  bnd     Z       C       [13:123457[C    6       PASS    SVTYPE=BND

==> 
##如event2表现为染色体易位

#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO
#event1
2       321681  bnd     W       G       G]17:198982]    6       PASS    SVTYPE=BND
17      198982  bnd     Y       A       A]2:321681]     6       PASS    SVTYPE=BND
#event2
2       321682  bnd     V       T       ]13:123456]T    6       PASS    SVTYPE=BND
13      123456  bnd     U       C       C[2:321682[     6       PASS    SVTYPE=BND
#event3
13      123457  bnd     X       A       [17:198983[A    6       PASS    SVTYPE=BND
17      198983  bnd     Z       C       [13:123457[C    6       PASS    SVTYPE=BND
```
![](pics/20200529.png)

```
14      38454918        5956_2  N       ]8:9174263]N    25.35   .       SVTYPE=BND;STRANDS=-+:4;CIPOS=-9,8;CIEND=-9,8;CIPOS95=0,0;CIEND95=0,0
8       9174263 5956_1  N       N[14:38454918[  25.35   .       SVTYPE=BND;STRANDS=+-:4;CIPOS=-9,8;CIEND=-9,8;CIPOS95=0,0;CIEND95=0,0;MATEID=....
```

BND即breakend的缩写。在CHROM、POS中展示第一个断点位置，在ALT中展示第二个断点位置，例如：A]10:115172011]、[12:70547434[C。通过比较第一个断点和第二个断点的染色体，可以判断是染色体间易位还是染色体内易位（这里为了描述方便，使用了“第一个断点”、“第二个断点”的描述方式，事实上两个断点并没有顺序之分）。
值得注意的是，ALT中方括号的方向在判断融合基因中有重要的作用。...]...]指易位序列在第一个断点位置的3'端，[...[...指易位序列在第一个断点位置的5‘端

https://zhuanlan.zhihu.com/p/87408242

## 插入片段
Differences	between	insert	size	and	inferred	insert	size can	be	used	to	detect	evidence	of	structural	variants,	including

+ 串联重复（Tandem Duplicate）
+ 易位（Translocation）
+ 缺失（Deletion)
+ Insertions
+ 倒位 (Inversion)
+ Inter-chromosomal	rearrangements
  
![](pics/20200527_sv3.png)
![](pics/20200527_sv1.png)
![](pics/20200525_3.png)
![](pics/20200525_sv1.png)
![](pics/20200525_sv2.png)
![](pics/20200525_sv3.png)
> Rausch T, et al. (2012) Bioinformatics


## 过滤
1. raw counts of paired and split reads supporting each allele.  The split read counts will not always be available because some calls will be imprecise, so the paired read count (SAMPLE/PR), might be the best starting point.



https://www.jianshu.com/p/a3f73991390d
https://www.jianshu.com/p/4c8e109f0e6a

 + low sequencing complexity (e.g. repetitive sequences，sequencing biases（GC-content bias and homopolymers)）
 + regions of high sequence similarity (e.g. segmental-duplicated regions). 

## benchmark data
https://wabi-wiki.scilifelab.se/display/SHGATG/Structural+variation+calling